### PR TITLE
Houdini fails to start because missing settings

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
This error happens when admins update core addon without updating Houdini addon.
which can result in an error in this Houdini hook `SetDefaultDisplayView` and some review publish plugins because an expected Houdini setting won't be available which will raise an error. 
 
An alternative solution to split the Addons which by design won't update the client code of Houdini without uploading the new addon.

## Additional Info
I don't know whether this PR should be merged or we should just wait till the addons be split. 
Maybe @iLLiCiTiT can tell

## Testing notes:
Launching Houdini with the latest develop while not uploading the new Houdini addon shouldn't raise errors. 
you can do so by using Ayon in dev mode and pull latest `develop`.

### CODE REVIEW

**Review by moonyuet — APPROVED** [review]
Tested successfully in Houdini.

File: client/ayon_core/hosts/houdini/hooks/set_default_display_and_view.py [hunk 1]
```diff
@@ -25,7 +25,12 @@ def execute(self):
             return
 
         houdini_color_settings = \
-            self.data["project_settings"]["houdini"]["imageio"]["workfile"]
+            self.data["project_settings"]["houdini"]["imageio"].get("workfile", {})
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/houdini/hooks/set_default_display_and_view.py:28]:
```suggestion
            self.data["project_settings"]["houdini"]["imageio"].get("workfile")
```

File: client/ayon_core/hosts/houdini/plugins/create/create_review.py [hunk 2]
```diff
@@ -18,8 +18,8 @@ class CreateReview(plugin.HoudiniCreator):
 
     def apply_settings(self, project_settings):
         super(CreateReview, self).apply_settings(project_settings)
-        color_settings = project_settings["houdini"]["imageio"]["workfile"]
-        if color_settings["enabled"]:
+        color_settings = project_settings["houdini"]["imageio"].get("workfile", {})
+        if color_settings and color_settings["enabled"]:
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/houdini/plugins/create/create_review.py:22]:
```suggestion
        if not color_settings.get("enabled"):
```

File: client/ayon_core/hosts/houdini/plugins/publish/validate_review_colorspace.py [hunk 3]
```diff
@@ -46,8 +46,8 @@ def apply_settings(cls, project_settings):
         apply_plugin_settings_automatically(cls, settings, logger=cls.log)
 
         # Add review color settings
-        color_settings = project_settings["houdini"]["imageio"]["workfile"]
-        if color_settings["enabled"]:
+        color_settings = project_settings["houdini"]["imageio"].get("workfile", {})
+        if color_settings and color_settings["enabled"]:
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/houdini/plugins/publish/validate_review_colorspace.py:50]:
```suggestion
        if not color_settings.get("enabled"):
```

File: client/ayon_core/hosts/houdini/plugins/publish/validate_review_colorspace.py [hunk 4]
```diff
@@ -46,8 +46,8 @@ def apply_settings(cls, project_settings):
         apply_plugin_settings_automatically(cls, settings, logger=cls.log)
 
         # Add review color settings
-        color_settings = project_settings["houdini"]["imageio"]["workfile"]
-        if color_settings["enabled"]:
+        color_settings = project_settings["houdini"]["imageio"].get("workfile", {})
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/houdini/plugins/publish/validate_review_colorspace.py:49]:
```suggestion
        # workfile settings added in '0.2.13'
        color_settings = project_settings["houdini"]["imageio"].get("workfile", {})
```

**Review by iLLiCiTiT — APPROVED** [review]

**Comment by MustafaJafar** [comment]
Hey @moonyuet ,  Could you review it again ?

**Review by moonyuet — APPROVED** [review]
Tested successfully in Houdini

**Pull request merged**